lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -6506,11 +6485,32 @@ SILValue ADContext::promoteToDifferentiableFunction(
     // have smaller capacity than `actualIndices`. We expect this logic to go
     // away when we support `@differentiable` partial apply.
     // if (actualIndices != desiredIndices) { // TODO: Re-enable.
+    auto extendedDesiredIndices = desiredIndices.parameters->extendingCapacity(
+        getASTContext(), actualIndices.parameters->getCapacity());
     if (actualIndices.source != desiredIndices.source ||
-        !actualIndices.parameters->equals(
-            desiredIndices.parameters->extendingCapacity(getASTContext(),
-                actualIndices.parameters->getCapacity()))) {
-      assert(actualIndices.parameters->isSupersetOf(desiredIndices.parameters));
+        !actualIndices.parameters->equals(extendedDesiredIndices)) {
+      // Check if underlying original function reference has been partially
+      // applied with arguments. If so, produce an error: parameter subset
+      // thunks do not yet support this case because partially applied arguments
+      // cannot be propagated to parameter subset thunks.
+      auto didPartiallyApplyArguments = [](SILValue original) {
+        while (auto *pai =
+                   peerThroughFunctionConversions<PartialApplyInst>(original)) {
+          if (pai->getNumArguments() > 0)
+            return true;
+          original = pai->getOperand(0);
+        }
+        return false;
+      };
+      if (didPartiallyApplyArguments(origFnOperand)) {
+        emitNondifferentiabilityError(
+            origFnOperand, invoker,
+            diag::
```

Comment:
Awkward line break. Is it necessary?

Reply:
Good catch, `clang-format` suggests no newline.